data_model/src/events/data/filters.rs

```diff
@@ -1,192 +1,466 @@
-//! This module contains `EventFilter` and entities for filter
+//! This module contains filters for data events.
+//!
+//! (almost) Each event in [`super::events`], there's two corresponding types in this module:
+//! - `*EventMatcher` - matches one event kind (e.g. [`super::events::AccountEvent::Created`] with [`AccountEventMatcher::ByCreated`])
+//! - `*EventFilter` - struct combining an optional id matcher and an optional event matcher
+//!
+//! The ones not having a filter are [`super::events::ConfigurationEvent`] and [`super::events::ExecutorEvent`] (TODO: why?).
 
 use core::fmt::Debug;
 
-use derive_more::Constructor;
 use iroha_data_model_derive::model;
 
 pub use self::model::*;
 use super::*;
 
-/// Filter for all events
-pub type DataEventFilter = FilterOpt<DataEntityFilter>;
-
 #[model]
 pub mod model {
     use super::*;
 
-    /// Optional filter. May pass all items or may filter them by `F`
-    ///
-    /// It's better than `Optional<F>` because `Optional` already has its own `filter` method and it
-    /// would be ugly to use fully qualified syntax to call `Filter::filter()` method on it.
-    /// Also `FilterOpt` variant names look better for filter needs
     #[derive(
-        Debug,
-        Clone,
-        PartialEq,
-        Eq,
-        PartialOrd,
-        Ord,
-        Decode,
-        Encode,
-        Deserialize,
-        Serialize,
-        IntoSchema,
+        Debug, Clone, PartialEq, Eq, FromVariant, Decode, Encode, Deserialize, Serialize, IntoSchema,
     )]
-    #[serde(untagged)] // Unaffected by #3330
-    pub enum FilterOpt<F> {
-        /// Accept all items that will be passed to `filter()` method
-        #[serde(with = "accept_all_as_string")]
-        AcceptAll,
-        /// Use filter `F` to choose acceptable items passed to `filter()` method
-        BySome(F),
+    pub enum DataEventFilter {
+        /// Matches any data events ([`DataEvent`])
+        ByAny,
+        /// Matches only [`PeerEvent`]s
+        ByPeer(PeerEventFilter),
+        /// Matches only [`DomainEvent`]s
+        ByDomain(DomainEventFilter),
+        /// Matches only [`AccountEvent`]s
+        ByAccount(AccountEventFilter),
+        /// Matches only [`AssetEvent`]s
+        ByAsset(AssetEventFilter),
+        /// Matches only [`AssetDefinitionEvent`]s
+        ByAssetDefinition(AssetDefinitionEventFilter),
+        /// Matches only [`TriggerEvent`]s
+        ByTrigger(TriggerEventFilter),
+        /// Matches only [`RoleEvent`]s
+        ByRole(RoleEventFilter),
+        // We didn't have filters for these events before the refactor. Should we?
+        // Configuration(ConfigurationEventFilter),
+        // Executor(ExecutorEventFilter),
```

Comment:
yes for `Executor` (I assume you meant `ByExecutor`), but not sure about `Configuration` (again I assume you meant `ByConfiguration`)

Reply:
do we have `Executor` and `Configuration` events?

Reply:
Yes, we do: https://github.com/hyperledger/iroha/blob/eca8569b155081249b0b81613693e7a89b903ec5/data_model/src/events/data/events.rs#L90-L93

Reply:
add both